tropohelper/services.py

```diff
@@ -99,6 +103,8 @@ def create_json_redshift_firehose_from_stream(stack, name, firehose_arn,
 def create_cloud_watch_logs_metric_filter(stack, name, log_group_name, filter_pattern,
                                           metric_namespace='LogMetrics', metric_value='1', metric_default_value=0.0):
     """Add a Cloud Watch logs metric filter pointing to an existing log group."""
+    log_group = create_log_group(stack, log_group_name);
```

Comment:
Why are you attempting to create the log group inside the metric filter?

Reply:
Why not? you just pass log group and it makes sure it's also created. Log group is for the metric or the firehose, it BELONGS to these, it's not shared with anything else. In the case of stream + firehose, I have 2 separate functions which caller has to call separately because you might want to hoop up the stream to elastic search or something else in addition to the redshift firehose. So I expose the create stream function so that caller has freedom to do that. But in the case of log group I don't see that very helpful.

Reply:
Do not create the log group in here.  It's unnecessary and will likely end up with an error.  The log group isn't directly associated with the filter, it's associated with the stream.  Furthermore, you should make this a reference instead of calling the variable when you actually tell the alert what log to point to.

Reply:
moved out